nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmProxyFacet.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2018-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.repository.helm.internal.content.recipe;
+
+import java.io.IOException;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.sonatype.nexus.repository.content.facet.ContentProxyFacetSupport;
+import org.sonatype.nexus.repository.view.Content;
+import org.sonatype.nexus.repository.view.Context;
+import org.sonatype.nexus.repository.view.matchers.token.TokenMatcher;
+import org.sonatype.repository.helm.internal.AssetKind;
+import org.sonatype.repository.helm.internal.content.HelmContentFacet;
+import org.sonatype.repository.helm.internal.util.HelmPathUtils;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+
+/**
+ * @since 1.0.10
+ */
+@Named
+public class HelmProxyFacet
+    extends ContentProxyFacetSupport
+{
+  private final HelmPathUtils helmPathUtils;
+
+  private static final String INDEX_YAML = "index.yaml";
+
+  @Inject
+  public HelmProxyFacet(final HelmPathUtils helmPathUtils)
+  {
+    this.helmPathUtils = checkNotNull(helmPathUtils);
+  }
+
+  @Override
+  protected Content getCachedContent(final Context context)  throws IOException {
+    return content().getAsset(getUrl(context)).orElse(null);
+  }
+
+  @Override
+  protected Content store(final Context context, final Content content) throws IOException {
+    AssetKind assetKind = context.getAttributes().require(AssetKind.class);
+    switch (assetKind) {
+      case HELM_INDEX:
+        return new Content(content().putIndex(getUrl(context), content, assetKind));
```

Comment:
You can just return the `Content` here

Reply:
yep, fixed 79e201777af2036e80b3e0cc895f9d22c8238ed7